### PR TITLE
pythonPackages.hiyapyco: init at 0.4.16

### PR DESCRIPTION
**Pull request by veehaitch**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/hiyapyco/default.nix [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  version = "0.4.16";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1qb7fb90jzpzx84adzwsabxq4hb5nc44szxn08nxkr2flzp1kvpz";
+  };
+
+  propagatedBuildInputs = [ pyyaml jinja2 ];
+
+  meta = with lib; {
```
Comment (jonringer) [inline pkgs/development/python-modules/hiyapyco/default.nix:14]: There are no tests included. Please try to checkout from source and check if they have unit tests, and try to run them. Unit tests give a good indication that they package has a high degree of validity and correctness given the python package set.

If tests are not available, then please use `pythonImportsCheck` to import the most important modules. This isn't as good as unit tests, but will usually give a good indication of run-time errors.

```

----------------------------------------------------------------------
Ran 0 tests in 0.000s
```

**Comment by jonringer** [comment]
I think i said this on the other PR, but thanks for creating your first PR with nixpkgs :)

File: pkgs/development/python-modules/hiyapyco/default.nix (hunk 1 above)
Comment (veehaitch) [reply pkgs/development/python-modules/hiyapyco/default.nix:26]: Good point. Apparently, the Pypi version does not include all the dependencies to run the test. I've switched to the source [from Github](https://github.com/zerwes/hiyapyco) and the tests are now executed. LGTM on Python 2 and 3.

**Comment by veehaitch** [comment]
> I think i said this on the other PR, but thanks for creating your first PR with nixpkgs :)

Thanks again :)

Strictly speaking, however, this was my first PR: https://github.com/NixOS/nixpkgs/pull/88793 😅

File: pkgs/development/python-modules/hiyapyco/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    rev = "release-${version}";
+    sha256 = "1ams9dp05yhgbg6255wrjgchl2mqg0s34d8b8prvql9lsh59s1fj";
+  };
+
+  propagatedBuildInputs = [ pyyaml jinja2 ];
+
+  checkPhase = ''
+    set -e
+    find test -name 'test_*.py' -exec python {} \;
+  '';
```
Comment (jonringer) [inline pkgs/development/python-modules/hiyapyco/default.nix:24]: ... wish they used a standard test runner like pytest... but i guess this does run tests

**Review by jonringer — APPROVED** [review]
LGTM

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 88871 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>3 packages built:</summary><br>- python27Packages.hiyapyco<br>- python37Packages.hiyapyco<br>- python38Packages.hiyapyco</details>

**Pull request merged**